pyaedt/modules/PostProcessor.py

```diff
@@ -601,7 +601,351 @@ def surfacePlotInstruction(self):
+        self.oreportsetup.PasteReports()
+        return True
+
+    @aedt_exception_handler
+    def delete_report(self, PlotName):
+        """Delete a field plot report.
+
+        Parameters
+        ----------
+        PlotName : str
+            Name of the field plot report.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+        """
+        self.oreportsetup.DeleteReports([PlotName])
+        return True
+
+    @aedt_exception_handler
+    def rename_report(self, PlotName, newname):
+        """Rename a plot.
+
+        Parameters
+        ----------
+        PlotName : str
+            Name of the plot.
+        newname : str
+            New name of the plot.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+        """
+        self.oreportsetup.RenameReport(PlotName, newname)
+        return True
+
+    @aedt_exception_handler
+    def get_report_data(self, expression="dB(S(1,1))", setup_sweep_name='', domain='Sweep', families_dict=None,
+                        report_input_type=None):
+        """Generate report data using the :func:`GetSolutionDataPerVariation` method.
+
+        This method returns the data object and the arrays ``solData`` and
+        ``FreqVals``.
+
+        Parameters
+        ----------
+        expression : str or list
+            One or more formulas to add to the report. The default is
+            ``"dB(S(1,1))"``.
+        setup_sweep_name : str, optional
+            Name of the setup for computing the report. The
+            default is ``""``, in which case the nominal sweep is
+            used.
+        domain : str or list, optional
+            Context type. The options are ``"Sweep"`` or
+            ``"Time"``. The default is ``"Sweep".``
+        families_dict : dict, optional
+            Dictionary of all families including the primary
+            sweep. The default is ``{"Freq": ["All"]}``.
+        report_input_type :  str
+            Type of input data for the report.
+
+        Returns
+        -------
+        :class:`pyaedt.modules.PostProcessor.SolutionData`
+
+        Examples
+        --------
+        Generate a report with the default sweep and default variation.
+
+        >>> hfss = HFSS()
+        >>> hfss.post.get_report_data("S(1,1)")
+
+        >>> m3d = Maxwell3D()
+        >>> m3d.post.get_report_data("SurfaceLoss")   # Eddy Current examples
+        >>> m3d.post.get_report_data("Wind(LoadA,LaodA)")    # TransientAnalsysis
+
+        """
+        if self.post_solution_type in ["3DLayout", "NexximLNA", "NexximTransient"]:
+            if domain == "Sweep":
+                did = 3
+            else:
+                did = 1
+            ctxt = ["NAME:Context", "SimValueContext:=",
+                    [did, 0, 2, 0, False, False, -1, 1, 0, 1, 1, "", 0, 0, "IDIID", False, "1"]]
+        elif type(domain) is list:
```

Comment:
It's generally prefered to use `isinstance` rather than type.  Here's a good blog post that discusses it: https://switowski.com/blog/type-vs-isinstance

```suggestion
        elif isinstance(domain, list):
```